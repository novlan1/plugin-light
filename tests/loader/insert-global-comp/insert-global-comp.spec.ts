import insertGlobalComponent from '../../../loader/insert-global-comp';

jest.mock('loader-utils', () => ({
  getOptions: jest.fn().mockReturnValue({
    pages: ['MOCK_PAGE'],
    components: [{
      isOnTop: false,
      name: 'MatchCommDialog', // 组件名称
      id: 'tip-match-comm-tips-dialog',
    }],
  }),
}));

describe('insertGlobalComponent', () => {
  it('only one tag', () => {
    const source = `
<template>
  <ModuleCustomGroupType
    :game-info-key="gameInfoKey"
  />
</template>
`;
    process.env.VUE_APP_PLATFORM = 'mp-weixin';
    expect(insertGlobalComponent.call({
      resourcePath: 'MOCK_PAGE',
    }, source)).toMatchSnapshot();
  });


  it('first tag is comment', () => {
    const source = `<template>
<!-- 战后数据和直播页面 -->
<div
  class="battle-detail-wrap"
  style="width:100%;height: 100%;"
>
  <page-loading v-if="mShowPageLoading" />
  <BattleDetailModule v-else-if="isScheEnd" />
  <VideoRoomModule v-else />
</div>
</template>`;
    process.env.VUE_APP_PLATFORM = 'mp-weixin';
    expect(insertGlobalComponent.call({
      resourcePath: 'MOCK_PAGE',
    }, source)).toMatchSnapshot();
  });


  it('already have 1', () => {
    const source = `<template>
<!-- 战后数据和直播页面 -->
<div
  class="battle-detail-wrap"
  style="width:100%;height: 100%;"
>
  <page-loading v-if="mShowPageLoading" />
  <BattleDetailModule v-else-if="isScheEnd" />
  <VideoRoomModule v-else />
  <MatchCommDialog id="tip-match-comm-tips-dialog" /></template>
</div>
</template>`;
    process.env.VUE_APP_PLATFORM = 'mp-weixin';
    expect(insertGlobalComponent.call({
      resourcePath: 'MOCK_PAGE',
    }, source)).toMatchSnapshot();
  });


  it('already have 2', () => {
    const source = `<template>
<!-- 战后数据和直播页面 -->
<div
  class="battle-detail-wrap"
  style="width:100%;height: 100%;"
>
  <page-loading v-if="mShowPageLoading" />
  <BattleDetailModule v-else-if="isScheEnd" />
  <VideoRoomModule v-else />
  <match-comm-dialog id="tip-match-comm-tips-dialog" /></template>
</div>
</template>`;
    process.env.VUE_APP_PLATFORM = 'mp-weixin';
    expect(insertGlobalComponent.call({
      resourcePath: 'MOCK_PAGE',
    }, source)).toMatchSnapshot();
  });
});


describe('Other Html Tag', () => {
  it('p', () => {
    const source = `<template>
<p
  class="battle-detail-wrap"
  style="width:100%;height: 100%;"
>
  <page-loading v-if="mShowPageLoading" />
  <BattleDetailModule v-else-if="isScheEnd" />
  <VideoRoomModule v-else />
</p>
</template>`;

    process.env.VUE_APP_PLATFORM = 'mp-weixin';
    expect(insertGlobalComponent.call({
      resourcePath: 'MOCK_PAGE',
    }, source)).toMatchSnapshot();
  });


  it('span', () => {
    const source = `<template>
<span
  class="battle-detail-wrap"
  style="width:100%;height: 100%;"
>
  <a>123</a>
</span>
</template>`;

    process.env.VUE_APP_PLATFORM = 'mp-weixin';
    expect(insertGlobalComponent.call({
      resourcePath: 'MOCK_PAGE',
    }, source)).toMatchSnapshot();
  });
});
