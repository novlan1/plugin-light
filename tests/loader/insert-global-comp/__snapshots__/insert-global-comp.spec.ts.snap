// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Other Html Tag p 1`] = `
"<template>
<p
  class="battle-detail-wrap"
  style="width:100%;height: 100%;"
><template></template>
  <page-loading v-if="mShowPageLoading" />
  <BattleDetailModule v-else-if="isScheEnd" />
  <VideoRoomModule v-else />
<template><MatchCommDialog id="tip-match-comm-tips-dialog" /></template></p>
</template>"
`;

exports[`Other Html Tag span 1`] = `
"<template>
<span
  class="battle-detail-wrap"
  style="width:100%;height: 100%;"
><template></template>
  <a>123</a>
<template><MatchCommDialog id="tip-match-comm-tips-dialog" /></template></span>
</template>"
`;

exports[`insertGlobalComponent already have 1 1`] = `
"<template>
<!-- 战后数据和直播页面 -->
<div
  class="battle-detail-wrap"
  style="width:100%;height: 100%;"
>
  <page-loading v-if="mShowPageLoading" />
  <BattleDetailModule v-else-if="isScheEnd" />
  <VideoRoomModule v-else />
  <MatchCommDialog id="tip-match-comm-tips-dialog" /></template>
</div>
</template>"
`;

exports[`insertGlobalComponent already have 2 1`] = `
"<template>
<!-- 战后数据和直播页面 -->
<div
  class="battle-detail-wrap"
  style="width:100%;height: 100%;"
>
  <page-loading v-if="mShowPageLoading" />
  <BattleDetailModule v-else-if="isScheEnd" />
  <VideoRoomModule v-else />
  <match-comm-dialog id="tip-match-comm-tips-dialog" /></template>
</div>
</template>"
`;

exports[`insertGlobalComponent first tag is comment 1`] = `
"<template>
<!-- 战后数据和直播页面 -->
<div
  class="battle-detail-wrap"
  style="width:100%;height: 100%;"
><template></template>
  <page-loading v-if="mShowPageLoading" />
  <BattleDetailModule v-else-if="isScheEnd" />
  <VideoRoomModule v-else />
<template><MatchCommDialog id="tip-match-comm-tips-dialog" /></template></div>
</template>"
`;

exports[`insertGlobalComponent only one tag 1`] = `
"
<template>
  <div><template></template> <ModuleCustomGroupType
    :game-info-key="gameInfoKey"
  /> <template><MatchCommDialog id="tip-match-comm-tips-dialog" /></template></div>
</template>
"
`;
